helix-lock/src/test/java/org/apache/helix/lock/TestZKHelixNonblockingLock.java

```diff
@@ -0,0 +1,176 @@
+/*
```

Comment:
Could you also test the case when two threads both try to acquire the lock, independently, and one of them succeed, and the other fail?

Reply:
Test added. Please see testSimultaneousAcquire().